### ISSUE TITLE
##[error]Unable to create ref when having multiple artifacts build from same repository

### ISSUE THREAD

**Issue by ggirard07** [jabbera/vsts-git-release-tag#20]
**Steps to reproduce**
- Create new release definition with the tag task.
- Add at least 2 artifact built from same repository with same commit.
- Create a new release instance and start it on any environment.

**Expected result**
A single tag is created and task complete successfully

**Actual result**
Tag is properly created but error logs cause the task to fail

**Description**
We have a release definition which deploy multiple artifacts all built from the same Git repository.
After enabling debug, it looks like the first artifact is properly tagged. Then other artifacts built from same commit fail as tag already exists.

```
2018-01-11T19:33:33.6320992Z Current agent version: '2.126.0'
...
2018-01-11T19:38:54.3176187Z ##[section]Starting: Tag Artifacts
2018-01-11T19:38:54.3181417Z ==============================================================================
2018-01-11T19:38:54.3181916Z Task         : Tag Artifacts
2018-01-11T19:38:54.3182362Z Description  : Tag artifacts during release with ReleaseName.
2018-01-11T19:38:54.3182781Z Version      : 3.0.5
2018-01-11T19:38:54.3183163Z Author       : Michael Barry
2018-01-11T19:38:54.3183538Z Help         : 
2018-01-11T19:38:54.3184899Z ==============================================================================
...
2018-01-11T18:41:17.9506411Z ##[debug]Processing artifact: 'artifact1' for ref: tags/my-release-0.0.0.0 new commit: 1e30618fd4249463bf9702fb04bca50b3cf95094 old commit 0000000000000000000000000000000000000000
2018-01-11T18:41:18.3577201Z ##[debug]Ref updated!
2018-01-11T18:41:18.3578930Z ##[debug]Processing artifact: 'artifact2' for ref: tags/my-release-0.0.0.0 new commit: 1e30618fd4249463bf9702fb04bca50b3cf95094 old commit 0000000000000000000000000000000000000000
2018-01-11T18:41:18.4281065Z ##[error]If you need to change permissions see: _admin/_versioncontrol?_a=security&repositoryId=00000000-0000-0000-0000-000000000000
2018-01-11T18:41:18.4282065Z ##[debug]Processed: ##vso[task.issue type=error;]If you need to change permissions see: _admin/_versioncontrol?_a=security&repositoryId=00000000-0000-0000-0000-000000000000
2018-01-11T18:41:18.4283183Z ##[debug]task result: Failed
2018-01-11T18:41:18.4285197Z ##[error]Unable to create ref: tags/my-release-0.0.0.0 UpdateStatus: 2 RepositoryId: 00000000-0000-0000-0000-000000000000 Commit: 1e30618fd4249463bf9702fb04bca50b3cf95094
2018-01-11T18:41:18.4286226Z ##[debug]Processed: ##vso[task.issue type=error;]Unable to create ref: tags/my-release-0.0.0.0 UpdateStatus: 2 RepositoryId: 00000000-0000-0000-0000-000000000000 Commit: 1e30618fd4249463bf9702fb04bca50b3cf95094
2018-01-11T18:41:18.4287967Z ##[debug]Processed: ##vso[task.complete result=Failed;]Unable to create ref: tags/my-release-0.0.0.0 UpdateStatus: 2 RepositoryId: 00000000-0000-0000-0000-000000000000 Commit: 1e30618fd4249463bf9702fb04bca50b3cf95094
2018-01-11T18:41:18.4289503Z ##[debug]Processing artifact: 'artifact3' for ref: tags/my-release-0.0.0.0 new commit: 1e30618fd4249463bf9702fb04bca50b3cf95094 old commit 0000000000000000000000000000000000000000
2018-01-11T18:41:18.6298119Z ##[error]If you need to change permissions see: _admin/_versioncontrol?_a=security&repositoryId=00000000-0000-0000-0000-000000000000
2018-01-11T18:41:18.6302369Z ##[debug]Processed: ##vso[task.issue type=error;]If you need to change permissions see: _admin/_versioncontrol?_a=security&repositoryId=00000000-0000-0000-0000-000000000000
2018-01-11T18:41:18.6303365Z ##[debug]task result: Failed
2018-01-11T18:41:18.6305111Z ##[error]Unable to create ref: tags/my-release-0.0.0.0 UpdateStatus: 2 RepositoryId: 00000000-0000-0000-0000-000000000000 Commit: 1e30618fd4249463bf9702fb04bca50b3cf95094
2018-01-11T18:41:18.6306533Z ##[debug]Processed: ##vso[task.issue type=error;]Unable to create ref: tags/my-release-0.0.0.0 UpdateStatus: 2 RepositoryId: 00000000-0000-0000-0000-000000000000 Commit: 1e30618fd4249463bf9702fb04bca50b3cf95094
2018-01-11T18:41:18.6308436Z ##[debug]Processed: ##vso[task.complete result=Failed;]Unable to create ref: tags/my-release-0.0.0.0 UpdateStatus: 2 RepositoryId: 00000000-0000-0000-0000-000000000000 Commit: 1e30618fd4249463bf9702fb04bca50b3cf95094
2018-01-11T18:41:18.6309751Z ##[debug]Processing artifact: 'artifact5' for ref: tags/my-release-0.0.0.0 new commit: c87386571016e778c06e0b486f076bd64f1f9b52 old commit 0000000000000000000000000000000000000000
2018-01-11T18:41:19.1303616Z ##[debug]Ref updated!
2018-01-11T18:41:19.1305377Z ##[debug]Processing artifact: 'artifact6' for ref: tags/my-release-0.0.0.0 new commit: 1e30618fd4249463bf9702fb04bca50b3cf95094 old commit 0000000000000000000000000000000000000000
2018-01-11T18:41:19.1810218Z ##[error]If you need to change permissions see: _admin/_versioncontrol?_a=security&repositoryId=00000000-0000-0000-0000-000000000000
2018-01-11T18:41:19.1811066Z ##[debug]Processed: ##vso[task.issue type=error;]If you need to change permissions see: _admin/_versioncontrol?_a=security&repositoryId=00000000-0000-0000-0000-000000000000
2018-01-11T18:41:19.1811997Z ##[debug]task result: Failed
2018-01-11T18:41:19.1813721Z ##[error]Unable to create ref: tags/my-release-0.0.0.0 UpdateStatus: 2 RepositoryId: 00000000-0000-0000-0000-000000000000 Commit: 1e30618fd4249463bf9702fb04bca50b3cf95094
2018-01-11T18:41:19.1814810Z ##[debug]Processed: ##vso[task.issue type=error;]Unable to create ref: tags/my-release-0.0.0.0 UpdateStatus: 2 RepositoryId: 00000000-0000-0000-0000-000000000000 Commit: 1e30618fd4249463bf9702fb04bca50b3cf95094
2018-01-11T18:41:19.1816671Z ##[debug]Processed: ##vso[task.complete result=Failed;]Unable to create ref: tags/my-release-0.0.0.0 UpdateStatus: 2 RepositoryId: 00000000-0000-0000-0000-000000000000 Commit: 1e30618fd4249463bf9702fb04bca50b3cf95094
2018-01-11T18:41:19.1895124Z ##[section]Finishing: Tag Artifacts
```
**Note:** artifact1, artifact2, artifact3, artifact4 and artifact6 are built from same repository commit. artifact5 is built from another repository.

After checking my repository, tag is properly created despite the error messages.
Also, If I redeploy the exact same release instance to same environment then I don't have any error messages. It seems the tag is properly detected and operation is aborted in that case:
```
2018-01-11T19:38:55.3876394Z ##[debug]Found matching ref for commit.
```

**Comment by jabbera** [comment]
This is fixed in 4.0.0 I hope. Didn't have time to test thoroughly hence the major upgrade.

**Comment by ggirard07** [comment]
@jabbera Thanks a lot for the fix, tested for my use case and seems to work pretty fine!
